packages/proxy/lib/http/util/prerequests.ts

```diff
@@ -157,8 +160,23 @@ export class PreRequests {
   }
 
   addPending (browserPreRequest: BrowserPreRequest) {
-    metrics.browserPreRequestsReceived++
     const key = `${browserPreRequest.method}-${tryDecodeURI(browserPreRequest.url)}`
+
+    // The initial request that loads the service worker does not always get sent to CDP. Thus, we need to explicitly ignore it. We determine
+    // it's the service worker request via the `service-worker` header
+    if (browserPreRequest.headers?.['Service-Worker'] === 'script') {
+      debugVerbose('Ignoring service worker script since we are not guaranteed to receive it', key)
```

Comment:
What happens if we do receive it?

Reply:
Updated with some clarification as to what is going on here: https://github.com/cypress-io/cypress/pull/28517/commits/956dd0aa92c8e7cdb0ee1284be92bdeeb52426a3